### PR TITLE
Increase test coverage and fix parse bug

### PR DESCRIPTION
**Pull request by night556**

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -411,60 +358,42 @@ function processMapArray(flagName, flag, input) {
...
-      let values = data.split(',');
-      for (let keyPair of values) {
-        let pair = keyPair.split('=');
-        tempValue[pair[0]] = pair[1];
-      }
-      value.push(tempValue);
+    let result;
+    switch (flag.subType) {
+    case 'string','number':
+      result['value']=input;
```
Comment (JacksonTian) [inline lib/parser.js:364]: 格式化弄好点。

File: lib/parser.js [hunk 2]
```diff
@@ -411,60 +358,42 @@ function processMapArray(flagName, flag, input) {
...
-      if (Object.keys(val).length !== Object.keys(mappingObj).length) {
-        let err = {
-          prompt: i18n.filedErr,
-          values: [flagName]
-        };
-        return { err };
-      }
-      mappingValue.push(mappingObj);
+    if (result.err&&result.err.prompt){
+      return {err:result.err};
```
Comment (JacksonTian) [inline lib/parser.js:371]: 同上

File: test/parser.test.js [hunk 3]
```diff
@@ -137,6 +138,85 @@ describe('parser.js', function () {
...
+    let parserCtx = parser.parser(ctx);
+    assert.strictEqual(parserCtx.err.prompt[ctx.profile.language], `选项 '%s' 的值json解析失败: %s`);
+    assert.deepStrictEqual(parserCtx.err.values, ['flag', 'Unexpected token v in JSON at position 1']);
+  });
+  it('string array type option analysis: json', function () {
+    ctx.args = ['array', '--flag', '["value1","value2"]'];
+    let parserCtx = parser.parser(ctx);
+    assert.deepStrictEqual(parserCtx.parsedValue, { 'flag': ['value1', 'value2'] });
+  });
+  it('number array type option analysis: vTypeMatchErr', function () {
```
Comment (JacksonTian) [inline test/parser.test.js:196]: 两个 it 之间的空行呢

**Review by JacksonTian — CHANGES_REQUESTED** [review]
SMC

**Comment by JacksonTian** [comment]
> This pull request **introduces 2 alerts** when merging [1e8d5fd](https://github.com/aliyun/alibabacloud-resourcify-cli/commit/1e8d5fd7869d3b8cb8a3d6f07c3b9caf603835fa) into [cef46c0](https://github.com/aliyun/alibabacloud-resourcify-cli/commit/cef46c0a3302dd634995ecf95d288419e6e805f6) - [view on LGTM.com](https://lgtm.com/projects/g/aliyun/alibabacloud-resourcify-cli/rev/pr-3b5d557dd8883f7f2d14ff6a9f628a1dafc0f7e9)
> 
> **new alerts:**
> 
> * 1 for Expression has no effect
> * 1 for Property access on null or undefined

有新问题

**Comment by night556** [comment]
> > This pull request **introduces 2 alerts** when merging [1e8d5fd](https://github.com/aliyun/alibabacloud-resourcify-cli/commit/1e8d5fd7869d3b8cb8a3d6f07c3b9caf603835fa) into [cef46c0](https://github.com/aliyun/alibabacloud-resourcify-cli/commit/cef46c0a3302dd634995ecf95d288419e6e805f6) - [view on LGTM.com](https://lgtm.com/projects/g/aliyun/alibabacloud-resourcify-cli/rev/pr-3b5d557dd8883f7f2d14ff6a9f628a1dafc0f7e9)
> > **new alerts:**
> > 
> > * 1 for Expression has no effect
> > * 1 for Property access on null or undefined
> 
> 有新问题

嗯 看到了，一并修改了

**Review by JacksonTian — APPROVED** [review]

**Pull request merged**